File: Transport/@stateflows/common/src/events/state-machine-info.ts
import { StateflowsEvent } from "../decorators/stateflows-event";
import { BehaviorInfo } from "./behavior-info";

export interface TreeNode<T> {
    value: T;
    nodes: Array<TreeNode<T>>;
}

export interface Tree<T> {
    value: T;
    root: TreeNode<T>;
}

@StateflowsEvent("Stateflows.StateMachines.StateMachineInfo, Stateflows.Common")
export class StateMachineInfo extends BehaviorInfo {
    public statesTree: Tree<string>;
}

File: Core/Stateflows.Common/StateMachines/Events/StateMachineInfo.cs
using Stateflows.Common;

namespace Stateflows.StateMachines
{
    [Retain, StrictOwnership]
    public class StateMachineInfo : BehaviorInfo
    {
        private StateMachineId id;

        public new StateMachineId Id
        {
            get => id;
            set
            {
                id = value;
                base.Id = value;
            }
        }
        
        public IReadOnlyTree<string> CurrentStates { get; set; }
    }
}
